.buildbot.sh

```diff
@@ -10,11 +10,19 @@ for example in *.c; do
     make bin/"${example%%.*}"
 done
 
-mkdir build 
-cd build
-    
-cmake -DCMAKE_TOOLCHAIN_FILE=riscv64-purecap.cmake ..
-make
+# arg-1 : Source directory
+# arg-2 : cmake toolchain file
+function build_cheri_examples()
+{
+    CWD=$(pwd)
```

Comment:
I'm not sure I quite see why we have `$1` (which is `$(pwd)`) and this?

Reply:
Thanks @ltratt . I just thought I could de-couple the source directory from the current working directory within the function.
In this case, the source directory and the current working directory just turned out to be the same, but the function can be reused in other directories. 

If it is over-engineered,easy enough to replace with $1.

Reply:
That is only true if one builds within the source tree, otherwise the source directory (`$1`) and the build root (`$(pwd)`) will be different. AFAIK the cmake convention is to always build "out-of-source" which is why there are two flags that one can pass to cmake. This can be seen below on line 22 where the `-B` flag gets the build root path, whereas the `-S` flag gets the source path.

Reply:
Unless I'm reading this incorrectly, `$CWD` is always `==` to `$1`. So I'm not sure we're getting the effect we expect.

Reply:
Changed everything within function to `${1}`.